fe/fe-core/src/main/java/org/apache/doris/analysis/TimeV2Literal.java

```diff
@@ -0,0 +1,269 @@
+
+    protected float hour;
+    protected long minute;
+    protected long second;
+    protected long microsecond;
+
+    /**
+     * C'tor forcing type, e.g., due to implicit cast
+     */
+    // for restore
+    private TimeV2Literal() {
+        this.type = Type.TIMEV2;
+        this.hour = 0;
+        this.minute = 0;
+        this.second = 0;
+        this.microsecond = 0;
+    }
+
+    public TimeV2Literal(float hour, long minute, long second) {
+        super();
+        this.type = Type.TIMEV2;
+        this.hour = hour;
+        this.minute = minute;
+        this.second = second;
+        this.microsecond = 0;
+        analysisDone();
+    }
+
+    public TimeV2Literal(float hour, long minute, long second, long microsecond, long scale) {
+        super();
+        this.type = ScalarType.createTimeV2Type((int) scale);
+        this.hour = hour;
+        this.minute = minute;
+        this.second = second;
+        this.microsecond = microsecond;
+        while (microsecond != 0 && this.microsecond < 100000) {
+            this.microsecond *= 10;
+        }
+        analysisDone();
+    }
+
+    public TimeV2Literal(String s) throws AnalysisException {
+        super();
+        init(s);
+        analysisDone();
+    }
+
+    protected TimeV2Literal(TimeV2Literal other) {
+        super(other);
+        this.type = ScalarType.createTimeV2Type(((ScalarType) other.type).getScalarScale());
+        this.hour = other.getHour();
+        this.minute = other.getMinute();
+        this.second = other.getSecond();
+        this.microsecond = other.getMicroSecond();
+    }
+
+    @Override
+    public Expr clone() {
+        return new TimeV2Literal(this);
+    }
+
+    protected void init(String s) throws AnalysisException {
+        // should like be/src/vec/runtime/time_value.h timev2_to_double_from_str
+        if (!s.contains(":")) {
+            boolean sign = false;
+            String tail = "";
+            if (s.charAt(0) == '-') {
+                s = s.substring(1);
+                sign = true;
+            }
+            if (s.contains(".")) {
+                tail = s.substring(s.indexOf("."));
+                s = s.substring(0, s.indexOf("."));
+            }
+            int len = s.length();
+            if (len == 1) {
+                s = "00:00:0" + s;
+            } else if (len == 2) {
+                s = "00:00:" + s;
+            } else if (len == 3) {
+                s = "00:0" + s.charAt(0) + ":" + s.substring(1);
+            } else if (len == 4) {
+                s = "00:" + s.substring(0, 2) + ":" + s.substring(2);
+            } else {
+                s = s.substring(0, len - 4) + ":" + s.substring(len - 4, len - 2) + ":" + s.substring(len - 2);
+            }
+            if (sign) {
+                s = '-' + s;
+            }
+            s = s + tail;
+        }
+        if (s.indexOf(':') == s.lastIndexOf(':')) {
```

Comment:
if `s` does not contain `':'`, if statement will be true, and then append `":00"` on `s`?